back/app/controllers/web_api/v1/users_controller.rb

```diff
@@ -110,6 +110,21 @@ def by_invite
     show
   end
 
+  # To validate an email without creating a user and return whether to show terms if they are have no password
+  def check
+    @user = User.find_by email: params[:email]
+    skip_authorization
+    if !User::EMAIL_REGEX.match?(params[:email])
+      render json: { errors: { email: [{ error: 'invalid', value: params[:email] }] } }, status: :unprocessable_entity
+    elsif @user && !@user&.no_password?
+      render json: { errors: { email: [{ error: 'taken', value: params[:email] }] } }, status: :unprocessable_entity
+    elsif @user&.registration_completed_at.present?
+      render json: { action: 'continue' }
```

Comment:
So if I get this right, this means that if a user already exists and has already verified this email, but does not have a password, we "continue," and "continue" probably means sending another verification code, right?